### PR TITLE
azure virtual machine facts -- querying all machines in subscription fix

### PR DESCRIPTION
**Pull request by zikalino**
##### SUMMARY
Fixing https://github.com/ansible/ansible/issues/53995

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
azure_rm_virtualmachine_facts

### CODE REVIEW

**Comment by Fred-sun** [comment]
@yungezz @yuwzho review_need

File: lib/ansible/modules/cloud/azure/azure_rm_virtualmachine_facts.py [hunk 1]
```diff
@@ -278,17 +280,30 @@ def list_items(self):
...
             if self.has_tags(item.tags, self.tags):
-                results.append(self.serialize_vm(self.get_vm(item.name)))
+                results.append(self.serialize_vm(self.get_vm(self.resource_group, item.name)))
         return results
 
-    def get_vm(self, name):
+    def list_all_items(self):
+        self.log('List all items')
+        try:
+            items = self.compute_client.virtual_machines.list_all()
```
Comment (yungezz) [inline lib/ansible/modules/cloud/azure/azure_rm_virtualmachine_facts.py:287]: is the response paged?

Comment (zikalino) [reply lib/ansible/modules/cloud/azure/azure_rm_virtualmachine_facts.py:287]: yes, handled by the standard python iterator

File: lib/ansible/modules/cloud/azure/azure_rm_virtualmachine_facts.py [hunk 2]
```diff
@@ -278,18 +277,31 @@ def list_items(self):
...
+        except CloudError as exc:
+            self.fail("Failed to list all items - {0}".format(str(exc)))
+
+        results = []
+        for item in items:
+            if self.has_tags(item.tags, self.tags):
+                results.append(self.get_vm(re.sub('\\/.*', '', re.sub('.*resourceGroups\\/', '', item.id)), item.name))
```
Comment (yungezz) [inline lib/ansible/modules/cloud/azure/azure_rm_virtualmachine_facts.py]: using existing parse_reosurce_id utility to get resource group

**Review by yungezz — APPROVED** [review]

**Pull request merged**